core/lib/pubs.ts

```diff
@@ -36,3 +40,44 @@ export const getPubTitle = (pub: PubTitleProps): string => {
 
 	return title ?? fallbackTitle;
 };
+
+type InputPub = ProcessedPub<{ withStage: true; withLegacyAssignee: true; withPubType: true }>;
+
+export const processedPubToPubResult = <T extends InputPub>(pub: T): GetPubsResult[number] => {
+	return {
+		...pub,
+		values: pub.values.reduce(
+			(acc, value) => {
+				const existingValue = acc[value.fieldSlug] as JsonValue | JsonValue[] | undefined;
+
+				if (!value?.relatedPubId) {
+					acc[value.fieldSlug] = value.value as JsonValue;
+					return acc;
+				}
+
+				const existingVal = existingValue
+					? Array.isArray(existingValue)
+						? existingValue
+						: [existingValue]
+					: [];
+
+				acc[value.fieldSlug] = [
+					...existingVal,
+					{ relatedPubId: value.relatedPubId, value: value.value as JsonValue },
+				];
+
+				return acc;
+			},
+			{} as GetPubsResult[number]["values"]
+		),
+		stages: pub.stage ? [pub.stage] : [],
+		assigneeId: pub.assignee?.id ?? null,
+		assignee: (pub.assignee ?? null) as GetPubsResult[number]["assignee"],
+		pubType: pub.pubType as GetPubsResult[number]["pubType"],
+		children: pub.children.length ? pub.children.map(processedPubToPubResult) : [],
+	};
+};
```

Comment:
this is a bridge function for places where we still use the `{ slug: value }` pubvalues shape, rather than an array
this is eg the case in the contexteditor, which i didn't feel like changing

Reply:
could you put this comment into the code? i'm definitely going to forget what the purpose of this is